src/audio/mixin_mixout/mixin_mixout_hifi5.c

```diff
@@ -0,0 +1,285 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2024 Intel Corporation. All rights reserved.
+//
+// Author: Andrula Song <xiaoyuan.song@intel.com>
+
+#include <sof/common.h>
+
+#include "mixin_mixout.h"
+
+#ifdef MIXIN_MIXOUT_HIFI5
+
+#if CONFIG_FORMAT_S16LE
+static void mix_s16(struct cir_buf_ptr *sink, int32_t start_sample, int32_t mixed_samples,
+		    const struct cir_buf_ptr *source,
+		    int32_t sample_count, uint16_t gain)
+{
+	int samples_to_mix, samples_to_copy, left_samples;
+	int n, nmax, i, m, left;
+	ae_int16x4 in_sample, in_sample1;
+	ae_int16x4 out_sample, out_sample1;
+	ae_int16x8 *in;
+	ae_int16x8 *out;
+	ae_valignx2 inu = AE_ZALIGN128();
+	ae_valignx2 outu1 = AE_ZALIGN128();
+	ae_valignx2 outu2 = AE_ZALIGN128();
+	/* cir_buf_wrap() is required and is done below in a loop */
+	ae_int16 *dst = (ae_int16 *)sink->ptr + start_sample;
+	ae_int16 *src = source->ptr;
+
+	assert(mixed_samples >= start_sample);
+	samples_to_mix = AE_MIN_32_signed(mixed_samples - start_sample, sample_count);
```

Comment:
What's the difference between AE_MIN_32_signed() and AE_MIN32()? I can find only the latter from HiFi5 DSP User's Guide?

Reply:
I got help from @ShriramShastry , he  found this from manual: "AE_MIN_32_signed is an internal proto intended for use by the Xtensa C/C++ compiler, and not recommended for use as intrinsics by the programmer. ". I think it's good to replace those with AE_MIN32 to avoid the code to break in some future tools update.